src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -196,19 +205,229 @@ Scalar<DataType> time_deriv_of_lapse(
+    const gsl::not_null<tnsr::I<DataType, SpatialDim, Frame>*> dt_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  if (UNLIKELY(get_size(get<0>(*dt_shift)) != get_size(get(lapse)))) {
+    *dt_shift = tnsr::I<DataType, SpatialDim, Frame>(get(lapse));
+  }
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    dt_shift->get(i) = -get(lapse) * pi.get(1, 0) *
+                       spacetime_unit_normal.get(0) *
+                       inverse_spatial_metric.get(i, 0);
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (size_t a = 0; a < SpatialDim + 1; ++a) {
+        if (a != 0 or j != 0) {
+          dt_shift->get(i) -= get(lapse) * pi.get(j + 1, a) *
+                              spacetime_unit_normal.get(a) *
+                              inverse_spatial_metric.get(i, j);
+        }
+        for (size_t k = 0; k < SpatialDim; ++k) {
+          dt_shift->get(i) += shift.get(j) * spacetime_unit_normal.get(a) *
+                              phi.get(j, k + 1, a) *
+                              inverse_spatial_metric.get(i, k);
+        }
+      }
+    }
+    dt_shift->get(i) *= get(lapse);
+  }
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::I<DataType, SpatialDim, Frame> time_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  tnsr::I<DataType, SpatialDim, Frame> dt_shift{};
+  GeneralizedHarmonic::time_deriv_of_shift<SpatialDim, Frame, DataType>(
+      make_not_null(&dt_shift), lapse, shift, inverse_spatial_metric,
+      spacetime_unit_normal, phi, pi);
+  return dt_shift;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void time_deriv_of_lower_shift(
+    const gsl::not_null<tnsr::i<DataType, SpatialDim, Frame>*> dt_lower_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  if (UNLIKELY(get_size(get<0>(*dt_lower_shift)) != get_size(get(lapse)))) {
+    *dt_lower_shift = tnsr::i<DataType, SpatialDim, Frame>(get(lapse));
+  }
+  // get \partial_0 N^j
+  auto inverse_spatial_metric = determinant_and_inverse(spatial_metric).second;
```

Comment:
These should also use Variables + not_null. If you like I can add the overload for the `determinant_and_inverse` functions, let me know :)